src/backend/distributed/operations/citus_tools.c

```diff
@@ -52,14 +57,61 @@ static void ExecuteCommandsAndStoreResults(StringInfo *nodeNameArray,
 										   bool *statusArray,
 										   StringInfo *resultStringArray,
 										   int commandCount);
-static bool ExecuteRemoteQueryOrCommand(char *nodeName, uint32 nodePort,
-										char *queryString, StringInfo queryResult);
+static bool ExecuteRemoteQueryOrCommand(char *nodeName, uint32 nodePort, const char *user,
+										const char *database, char *queryString,
+										StringInfo queryResultString,
+										bool forceNewConnection);
 static Tuplestorestate * CreateTupleStore(TupleDesc tupleDescriptor,
 										  StringInfo *nodeNameArray, int *nodePortArray,
 										  bool *statusArray,
 										  StringInfo *resultArray, int commandCount);
 
 
+/*
+ * check_connection_to_node sends a simple query from a worker node to another
+ * node, and returns success status.
+ */
+Datum
+check_connection_to_node(PG_FUNCTION_ARGS)
+{
+	char *nodeName = PG_GETARG_TEXT_TO_CSTRING(0);
+	int32 nodePort = PG_GETARG_INT32(1);
+	char *userName = PG_GETARG_TEXT_TO_CSTRING_OR_NULL(2);
```

Comment:
One thing that I have been thinking and couldn't conclude yet is whether we should have the user/db parameters here.

And, my main concern if there are any edge cases that'd create a case where a user consumes resources that it shouldn't.

For example, `userName` does not have login right to the coordinator, but has rights to the worker. Should it be allowed that the coordinator use that user to connect to the worker? In the end, that user shouldn't be able to do anything on the coordinator. 

Given that and possibly some other edge cases like this, I'd probably simply drop username/db to avoid such cases.

In the end, our primary user (HSC) can connect with regular or super user. If needed in the future, we can expose these parameters after careful thinking?

These are probably unlikely edge cases, but given that we don't have to expose these parameters, I'd be in favor of dropping those.

Would be nice to get @marcocitus perspective on this as well just in case I miss something important here.

Reply:
As we checked with him on a private chat, he is also fine with dropping the parameters. @hanefi Let me know if you think differently on this subject

Reply:
I dropped the parameters and now the API is much simpler.